ext/ManoptJuMPExt.jl

```diff
@@ -0,0 +1,408 @@
+    function eval_f_cb(M, x)
+        if model.sense == MOI.FEASIBILITY_SENSE
+            return 0.0
+        end
+        obj = MOI.eval_objective(evaluator, JuMP.vectorize(x, _shape(model.manifold)))
+        if model.sense == MOI.MAX_SENSE
+            obj = -obj
+        end
+        return obj
+    end
+    function eval_grad_f_cb(M, X)
+        x = JuMP.vectorize(X, _shape(model.manifold))
+        grad_f = zeros(length(x))
+        if model.sense == MOI.FEASIBILITY_SENSE
+            grad_f .= zero(eltype(grad))
+        else
+            MOI.eval_objective_gradient(evaluator, grad_f, x)
+        end
+        if model.sense == MOI.MAX_SENSE
+            LinearAlgebra.rmul!(grad_f, -1)
+        end
+        reshaped_grad_f = JuMP.reshape_vector(grad_f, _shape(model.manifold))
+        return ManifoldDiff.riemannian_gradient(model.manifold, X, reshaped_grad_f)
+    end
+    mgo = Manopt.ManifoldGradientObjective(eval_f_cb, eval_grad_f_cb)
+    dmgo = decorate_objective!(model.manifold, mgo)
+    model.problem = DefaultManoptProblem(model.manifold, dmgo)
+    reshaped_start = JuMP.reshape_vector(start, _shape(model.manifold))
+    descent_state_type = model.options[DESCENT_STATE_TYPE]
+    kws = Dict{Symbol,Any}(
+        Symbol(key) => value for (key, value) in model.options if key != DESCENT_STATE_TYPE
+    )
+    s = descent_state_type(model.manifold, reshaped_start; kws...)
+    model.state = decorate_state!(s)
+    solve!(model.problem, model.state)
+    return nothing
+end
+
+struct ArrayShape{N} <: JuMP.AbstractShape
+    size::NTuple{N,Int}
+end
+
+function JuMP.vectorize(array::Array{T,N}, ::ArrayShape{M}) where {T,N,M}
+    return vec(array)
+end
+
+function JuMP.reshape_vector(vector::Vector, shape::ArrayShape)
+    return reshape(vector, shape.size)
+end
+
+function _shape(m::ManifoldsBase.AbstractManifold)
+    return ArrayShape(ManifoldsBase.representation_size(m))
+end
+
+function JuMP.build_variable(::Function, func, m::ManifoldsBase.AbstractManifold)
+    shape = _shape(m)
+    return JuMP.VariablesConstrainedOnCreation(
+        JuMP.vectorize(func, shape), VectorizedManifold(m), shape
+    )
+end
+
+function MOI.get(model::Optimizer, ::MOI.TerminationStatus)
+    if isnothing(model.state)
+        return MOI.OPTIMIZE_NOT_CALLED
+    else
+        return MOI.LOCALLY_SOLVED
+    end
+end
+
+function MOI.get(model::Optimizer, ::MOI.ResultCount)
+    if isnothing(model.state)
+        return 0
+    else
+        return 1
+    end
+end
+
+function MOI.get(model::Optimizer, ::MOI.PrimalStatus)
+    if isnothing(model.state)
+        return MOI.NO_SOLUTION
+    else
+        return MOI.FEASIBLE_POINT
+    end
+end
+
+MOI.get(::Optimizer, ::MOI.DualStatus) = MOI.NO_SOLUTION
+
+function MOI.get(model::Optimizer, ::MOI.RawStatusString)
+    # `strip` removes the `\n` at the end and returns an `AbstractString`
+    # Since MOI wants a `String`, we pass it through `string`
+    return string(strip(get_reason(model.state)))
```

Comment:
While first of all a short documentation would be nice.
Why are the `\n`s stripped?
Besides that, it also depends on what you expect / mean by `RawStatus`, `get_reason` of a state as is called here, would just return the reason the solver stopped, see https://manoptjl.org/stable/plans/stopping_criteria/#Manopt.get_reason-Tuple{AbstractManoptSolverState}.
the `repr(model.state)` would provide the full status/state of the solver after (also during) a run – but is also much more verbose and prints quite a few lines.

also `get_reason` should always return a string - but strip maybe does not?

Reply:
Yes, `strip` is a `Substring` so we need to convert to `String`. `RawStatusString` should be a short sentence (without an ending `\n` so `get_reason` is appropriate

Reply:
Ok.